File: src/main/java/net/dontdrinkandroot/gitki/wicket/event/FileMovedEvent.java
package net.dontdrinkandroot.gitki.wicket.event;

import net.dontdrinkandroot.gitki.model.FilePath;
import org.apache.wicket.ajax.AjaxRequestTarget;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class FileMovedEvent extends FileEvent
{
    private FilePath targetPath;

    public FileMovedEvent(FilePath filePath, AjaxRequestTarget target, FilePath targetPath)
    {
        super(filePath, target);
        this.targetPath = targetPath;
    }

    public FilePath getTargetPath()
    {
        return this.targetPath;
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/event/FileMovedEvent.kt
package net.dontdrinkandroot.gitki.wicket.event

import net.dontdrinkandroot.gitki.model.FilePath
import org.apache.wicket.ajax.AjaxRequestTarget

class FileMovedEvent(filePath: FilePath, target: AjaxRequestTarget?, val targetPath: FilePath) :
    FileEvent(filePath, target)